test/ectest.c

```diff
@@ -188,1185 +146,1033 @@ static void group_order_tests(EC_GROUP *group)
-    BN_print_fp(stdout, y);
-    fprintf(stdout, "\n");
+        || !TEST_false(EC_POINT_set_affine_coordinates_GFp(group, P, x,
+                                                          yplusone, ctx))
+        || !TEST_true(EC_POINT_set_affine_coordinates_GFp(group, P, x, y, ctx))
+        || !TEST_int_gt(EC_POINT_is_on_curve(group, P, ctx), 0)
+        || !TEST_true(BN_hex2bn(&z,
+                                "0100000000000000000001F4C8F927AED3CA752257"))
+        || !TEST_true(EC_GROUP_set_generator(group, P, z, BN_value_one()))
+        || !TEST_true(EC_POINT_get_affine_coordinates_GFp(group, P, x, y, ctx)))
+        goto err;
+    BIO_printf(bio_out, "\nSEC2 curve secp160r1 -- Generator:\n     x = 0x");
+    BN_print(bio_out, x);
+    BIO_printf(bio_out, "\n     y = 0x");
+    BN_print(bio_out, y);
+    BIO_printf(bio_out, "\n");
     /* G_y value taken from the standard: */
-    if (!BN_hex2bn(&z, "23a628553168947d59dcc912042351377ac5fb32"))
-        ABORT;
-    if (0 != BN_cmp(y, z))
-        ABORT;
-
-    fprintf(stdout, "verify degree ...");
-    if (EC_GROUP_get_degree(group) != 160)
-        ABORT;
-    fprintf(stdout, " ok\n");
-
-    group_order_tests(group);
-
-    if ((P_160 = EC_GROUP_new(EC_GROUP_method_of(group))) == NULL)
-        ABORT;
-    if (!EC_GROUP_copy(P_160, group))
-        ABORT;
+    if (!TEST_true(BN_hex2bn(&z, "23a628553168947d59dcc912042351377ac5fb32"))
+        || !TEST_int_eq(0, BN_cmp(y, z))
+        || !TEST_int_eq(EC_GROUP_get_degree(group), 160)
+        || !group_order_tests(group)
+        || !TEST_ptr(P_160 = EC_GROUP_new(EC_GROUP_method_of(group)))
+        || !TEST_true(EC_GROUP_copy(P_160, group))
 
     /* Curve P-192 (FIPS PUB 186-2, App. 6) */
 
-    if (!BN_hex2bn(&p, "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFF"))
-        ABORT;
-    if (1 != BN_is_prime_ex(p, BN_prime_checks, ctx, NULL))
-        ABORT;
-    if (!BN_hex2bn(&a, "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFFFFFFFFFC"))
-        ABORT;
-    if (!BN_hex2bn(&b, "64210519E59C80E70FA7E9AB72243049FEB8DEECC146B9B1"))
-        ABORT;
-    if (!EC_GROUP_set_curve_GFp(group, p, a, b, ctx))
-        ABORT;
-
-    if (!BN_hex2bn(&x, "188DA80EB03090F67CBF20EB43A18800F4FF0AFD82FF1012"))
-        ABORT;
-    if (!EC_POINT_set_compressed_coordinates_GFp(group, P, x, 1, ctx))
-        ABORT;
-    if (EC_POINT_is_on_curve(group, P, ctx) <= 0)
-        ABORT;
-    if (!BN_hex2bn(&z, "FFFFFFFFFFFFFFFFFFFFFFFF99DEF836146BC9B1B4D22831"))
-        ABORT;
-    if (!EC_GROUP_set_generator(group, P, z, BN_value_one()))
-        ABORT;
-
-    if (!EC_POINT_get_affine_coordinates_GFp(group, P, x, y, ctx))
-        ABORT;
-    fprintf(stdout, "\nNIST curve P-192 -- Generator:\n     x = 0x");
-    BN_print_fp(stdout, x);
-    fprintf(stdout, "\n     y = 0x");
-    BN_print_fp(stdout, y);
-    fprintf(stdout, "\n");
+        || !TEST_true(BN_hex2bn(&p, "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFFFF"
+                                    "FFFFFFFF"))
```

Comment:
Since we're here I'd like to suggest to break hex strings like that at multiples of 16, as opposite to honoring line length alone. I mean if we take it for a rule it would be much easier to recognize context. For example if above line was broken as
```
"FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFE"
"FFFFFFFFFFFFFFFF"
```
it would be much easier to recognize 192-bit number as well as specific pattern.

In the context one can also wonder if it would appropriate to write `"23a628553168947d59dcc912042351377ac5fb32"` as `"23a62855" "3168947d59dcc912042351377ac5fb32"` (single-line here) and `"0100000000000000000001F4C8F927AED3CA752257"` as `"0100000000" "000000000001F4C8F927AED3CA752257"` (multiple lines here). Or in other words favouring multiples of 16 from least significant part...